core/src/main/scala/me/jeffmay/neo4j/client/cypher/CypherArg.scala

```diff
@@ -0,0 +1,119 @@
+package me.jeffmay.neo4j.client.cypher
+
+import scala.language.implicitConversions
+import scala.util.{Failure, Success, Try}
+
+/**
+  * A container for the success or failure to build a Cypher argument.
+  *
+  * This avoids throwing an exception on the construction of a [[CypherArg]], but captures the exception
+  * at the point it occurs so that when you attempt to use the argument in the interpolated Cypher query,
+  * you see an exception with a stack trace that points to the where the failure happened.
+  */
+sealed trait CypherResult[+T <: CypherArg] {
+
+  def isValid: Boolean
+  @inline final def isInvalid: Boolean = !isValid
+
+  def getOrThrow: T
+  @inline final def tryGet: Try[T] = Try(getOrThrow)
+}
+object CypherResult {
+  implicit def valid[T <: CypherArg](arg: T): CypherResultValid[T] = CypherResultValid(arg)
+}
+
+/**
+  * Contains a valid [[CypherArg]] that can be used
+ *
+  * @param result the valid result.
+  */
+case class CypherResultValid[+T <: CypherArg](result: T) extends CypherResult[T] {
+  override def isValid: Boolean = true
```

Comment:
I could make this a `final def`. I don't want to allocate space for this value, when it isn't called that often. Also, static dispatch to a `def` with a constant is quick.